### DISCUSSION TITLE
How to add unit price and total price when printing purchase order as pdf?

### DISCUSSION THREAD

**Discussion by anonkwe** [inventree/InvenTree#4439] [Q&A] [answered]
How to add unit price and total price when printing purchase order as pdf?

{% decimal line.purchase_price %} does not work... not much seems to be  working

**Comment by SchrodingersGat** [comment, marked as answer]
Can you share your entire template?

**Reply by anonkwe** [reply]
```html
<h3 style="text-transform:uppercase;">{% trans "Items" %}</h3>
<table class='table table-striped table-condensed'>
    <thead>
        <tr>
		<th>Description</th>
		<th>QTEE/QTY</th>
		<th>Prix/Price</th>
		<th>Total</th>
            
        </tr>
    </thead>
    <tbody>
        {% for line in lines.all %}
        <tr>
            <td>
                <div class='thumb-container'>
                    <img src='{% part_image line.part.part %}' class='part-thumb'>
                </div>
                <div class='part-text'>
                    {{ line.part.part.full_name }}
                
            </td>
            <td>{% decimal line.quantity %} PMP</td>
			<td>{{ line.purchase_price }} {{ line.purchase_price_currency}} </td>
			<td>{{ line.total_price }} {{ line.purchase_price_currency}} </td>
		</tr>

  {% endfor %}
  		</div>
    </tbody>
</table>

{% endblock %}
```

**Reply by SchrodingersGat** [reply]
Seems to work fine on my end:

![image](https://user-images.githubusercontent.com/10080325/222638385-087f7dae-59cd-4ba0-b2bc-070e83502d9f.png)

I had to tweak your example to get it to work:

```html
{% extends "report/inventree_po_report.html" %}

{% load i18n %}
{% load inventree_extras %}
{% load report %}
{% load barcode %}

{% block page_content %}

<h3 style="text-transform:uppercase;">{% trans "Items" %}</h3>
<table class='table table-striped table-condensed'>
    <thead>
        <tr>
		<th>Description</th>
		<th>QTEE/QTY</th>
		<th>Prix/Price</th>
		<th>Total</th>
            
        </tr>
    </thead>
    <tbody>
        {% for line in lines.all %}
        <tr>
            <td>
                <div class='thumb-container'>
                    <img src='{% part_image line.part.part %}' class='part-thumb'>
                </div>
                <div class='part-text'>
                    {{ line.part.part.full_name }}
                
            </td>
            <td>{% decimal line.quantity %} PMP</td>
			<td>{{ line.purchase_price }} {{ line.purchase_price_currency}} </td>
			<td>{{ line.total_price }} {{ line.purchase_price_currency}} </td>
		</tr>

  {% endfor %}
  		</div>
    </tbody>
</table>

{% endblock %}
```

**Comment by anonkwe** [comment]
it ain't working for you as well, i wanna display this in  the  po:[image:
image.png]

**Reply by SchrodingersGat** [reply]
Do you mean that the "total price" is not rendering correctly?

The PurchaseOrderLineItem model does not have a method called *total_price* which is why it is not showing anything. We would need to add that functionality to allow rendering here

**Reply by SchrodingersGat** [reply]
Ref: https://github.com/inventree/InvenTree/pull/4444

**Reply by anonkwe** [reply]
still not working, is there anything i may have missed? It returns a no data
![image](https://user-images.githubusercontent.com/61838766/223198369-746ab9d4-988f-4d09-ab7f-dedd6be28f8c.png)
<td>{% include "price_data.html" with price=line.total_line_price %}</td>

**Reply by SchrodingersGat** [reply]
That's strange, it should be working based on that. Can you show what happens when you use the [inventree_po_report_base.html template](https://github.com/inventree/InvenTree/blob/master/InvenTree/report/templates/report/inventree_po_report_base.html)?

**Reply by SchrodingersGat** [reply]
Oh and it will only work if you have updated your installation to the latest master version, where I've added the `total_line_price` method

**Reply by anonkwe** [reply]
i'm quite new to this, how do i make sure i update to the latest version? if i do invoke update while on my env, it says master has no commmit or something like that?

**Reply by anonkwe** [reply]
{% extends "report/inventree_report_base.html" %}

{% load i18n %}
{% load report %}
{% load barcode %}
{% load inventree_extras %}

{% block page_margin %}
margin: 2cm;
margin-top: 4cm;
{% endblock %}

{% block bottom_left %}
content: "v{{report_revision}} - {{ date.isoformat }}";
{% endblock %}

{% block bottom_center %}
content: "{% inventree_version shortstring=True %}";
{% endblock %}

{% block style %}

.header-right {
    text-align: right;
    float: right;
}

.logo {
    height: 20mm;
    vertical-align: middle;
}

.thumb-container {
    width: 32px;
    display: inline;
}


.part-thumb {
    max-width: 32px;
    max-height: 32px;
    display: inline;
}

.part-text {
    display: inline;
}

table {
    border: 1px solid #eee;
    border-radius: 3px;
    border-collapse: collapse;
    width: 100%;
    font-size: 80%;
}

table td {
    border: 1px solid #eee;
}

table td.shrink {
    white-space: nowrap
}

table td.expand {
    width: 99%
}

{% endblock %}

{% block header_content %}

    <img class='logo' src='{% company_image supplier %}' alt="{{ supplier }}" width='150'>

    <div class='header-right'>
        <h3>{% trans "Purchase Order" %} {{ prefix }}{{ reference }}</h3>
        {% if supplier %}{{ supplier.name }}{% else %}{% trans "Supplier was deleted" %}{% endif %}
    </div>

{% endblock %}

{% block page_content %}

<h3>{% trans "Line Items" %}</h3>

<table class='table table-striped table-condensed'>
    <thead>
        <tr>
            <th>{% trans "Part" %}</th>
            <th>{% trans "Reference" %}</th>
            <th>{% trans "Quantity" %}</th>
            <th>{% trans "Unit Price" %}</th>
            <th>{% trans "Total Price" %}</th>
            <th>{% trans "Note" %}</th>
        </tr>
    </thead>
    <tbody>
        {% for line in lines.all %}
        <tr>
            <td>
                <div class='thumb-container'>
                    <img src='{% part_image line.part.part %}' class='part-thumb'>
                </div>
                <div class='part-text'>
                    {{ line.part.part.full_name }}
                </div>
            </td>
            <td>{{ line.reference }}</td>
            <td>{% decimal line.quantity %}</td>
            <td>{% include "price_data.html" with price=line.purchase_price %}</td>
            <td>{% include "price_data.html" with price=line.total_line_price %}</td>
            <td>{{ line.notes }}</td>
        </tr>
        {% endfor %}

        {% if extra_lines %}
        <tr><th colspan='6'>{% trans "Extra Line Items" %}</th></tr>
        {% for line in extra_lines.all %}
        <tr>
            <td></td>
            <td>{{ line.reference }}</td>
            <td>{% decimal line.quantity %}</td>
            <td>{% include "price_data.html" with price=line.price %}</td>
            <td>{% include "price_data.html" with price=line.total_line_price %}</td>
            <td>{{ line.notes }}</td>
        </tr>
        {% endfor %}
        {% endif %}

        <tr>
            <td></td>
            <td></td>
            <td></td>
            <th>{% trans "Total" %}</th>
            <td>{% include "price_data.html" with price=order.total_price %}</td>
            <td></td>
        </tr>

    </tbody>
</table>


{% endblock %}

**Reply by anonkwe** [reply]
its okay it's working, i did not pull the master branch the proper way the first time, now everything's fine

**Comment by anonkwe** [comment]
is there a  way to multiply  the unit price by the quantity  directly  on
the  report's html file

**Reply by SchrodingersGat** [reply]
Not currently - although I can see that this would be a useful addition! 

See incoming changes in https://github.com/inventree/InvenTree/pull/4443

**Comment by anonkwe** [comment]
is there a way to change decimals only on PO reports, if I put pricing decilmals to 2, it messes up all the table views of the site, decimals should be 2 on the report and 4 on the whole site. right now reports are taking in account 4 decimals:

![image](https://user-images.githubusercontent.com/61838766/223610814-eeba7a75-5be3-430c-a063-4a97b6a80b2c.png)

**Reply by SchrodingersGat** [reply]
@anonkwe check out new PR - https://github.com/inventree/InvenTree/pull/4469 - I've exposed the new method `render_currency` which will allow you to do this.

**Reply by SchrodingersGat** [reply]
https://docs.inventree.org/en/latest/report/helpers/#rendering-currency
